src/loader/module.js

```diff
@@ -0,0 +1,151 @@
+import ResolvablePromise from './resolvable-promise';
+
+/**
+ * A module descriptor
+ */
+export default class Module {
+	/**
+	 * @param {string} name name of module
+	 */
+	constructor(name) {
+		this._name = name;
+		this._dependencies = undefined;
+		this._factory = undefined;
+		this._implementation = undefined;
+
+		this._state = {
```

Comment:
I did that so that debugging the list of modules in the browser console is cleaner. A lot of times you console.log a list of modules and it is useful to be able to see just the important info (name, implementation, and dependencies). 

Think of it as a namespace.

Reply:
We have a debug, a production, and a minified build. Maybe we could make use of them and optimize for debuggin only in the first one.

However, to do that we would have to make changes to the portal, because there's no way to choose which one to use (the portal always used the production one) :-(.

We'll leave it like this and if it is a problem in the future, we may let users configure which version should be used. Nevertheless, I don't think this is going to pose any problem (apart from the use of RAM) because module objects stay forever once they are created, so they won't ever be GCed.

Reply:
> […] (the portal always used the production one) :-(.

This is a critical problem we’ve been facing from time to time. (Last discussion about it was just yesterday with @crcarlo). The fact that you can’t tell the portal to be in “dev” or “prod” env is a big PITA — _at least_ this should be possible to populate `process.env.NODE_ENV`.